coldfront/core/user/views.py

```diff
@@ -899,6 +919,41 @@ def get_form_kwargs(self):
     def get_success_url(self):
         return reverse('user-profile')
 
+class UpdateDepartmentsView(LoginRequiredMixin, FormView):
+
+    form_class = DepartmentSelectionForm
+    template_name = 'user/user_update_departments.html'
+    login_url = '/'
+
+    error_message = 'Unexpected failure. Please contact an administrator.'
+
+    def form_valid(self, form):
+        user = self.request.user
+        form_data = form.cleaned_data
+        new_departments = form_data['departments']
+        userprofile = user.userprofile
+        for department in new_departments:
+            UserDepartment.objects.get_or_create(userprofile=userprofile,
```

Comment:
I'm assuming the default for `is_authoritative` is `False`, but let's set it explicitly here.